activityMonitor.cjs

```diff
@@ -0,0 +1,48 @@
+const os = require('os');
+const childProcess = require('child_process');
+const fs = require('fs');
+
+const platform = os.platform();
+
+let result = '';
+
+const command = platform === 'win32' ?
+    `powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + ' ' + $_.CPU + ' ' + $_.WorkingSet }"` :
+    'ps -A -o %cpu,%mem,comm | sort -nr | head -n 1';
+
+const execProcess = (command, args) => {
+    const process = childProcess.exec(command, args);
+    
+    process.stderr.on('data', (data) =>  {
+        console.error(`stderr: ${data}`);
+    });
+
+    process.stdout.on('data', (data) => {
+        result = data;
+    });
+};
+
+const logWrite = (command) => {
+    execProcess(command);
+    const lineToAppend = `${Date.now()} : ${result}`;
+    fs.appendFile('activityMonitor.log', lineToAppend, (err) => {
+        if (err) console.log("And error has happened while reading/writing the file");
+    });
+};
+
+const consoleWrite = (command) => {
+    execProcess(command);
```

Comment:
You should make use of callbacks or even better of `Promise` to ensure the result you are using is the right one for the `execProcess` execution. Relying on the  global variable `result` makes your code susceptible to bugs. This is related to what we study in the previous module.

This also applies for the `logWrite` function.

Reply:
Hi @carboleda.
I've applied the changes you suggested, now I'm using a callback instead the global variable.
thanks